### ISSUE TITLE
Successful response data getting silently tossed due to a JSON parse error?

### ISSUE THREAD

**Issue by fairps** [axios/axios#1516]
#### Summary
We ran into an issue I thought I'd let y'all know about in case it turns out to be a bug.  I am making a call:
```
axios.post('/foo/.svc/r/GetHappyStuff', {
  <arguments go here>
},{
  cancelToken: cancelToken.token
}).then((resp) => {
  if (resp.data) {
    const res = resp.data.GetHappyResult
    if (res && res.succeeded) {
      // successful
    } 
  }
}).catch((err) => {
  console.error(err);
});
```

Looking in the Network tab of Chrome's dev tools I can see that the call completes successfully, and I can see the result in the Preview & Response tabs:
![image](https://user-images.githubusercontent.com/27831128/39599071-34146854-4ee0-11e8-8f65-c75b3b9bb777.png)
![image](https://user-images.githubusercontent.com/27831128/39599166-8accc704-4ee0-11e8-9e12-771011b3b655.png)

So the weird thing is that in the .then, I can write **resp** out the console and see the data.  I can write out **resp.data** and see the same, and the **if (resp.data)** evaluates to true.  As soon as I try and access the **.GetHappyResult** in any way, such as setting up the res variable, it will suddenly have a value of _undefined_.

What we have determined through pain / trial / error is that the contents of GetHappyResult are failing to parse through the JSON parser because there are some values which got written as NaN which causes the parser to barf... but it looks like that is being eaten and we happily try and use the value down the road where things fail.  

Anyway, we've fixed this but I wanted to share our experience since this has been pretty painful and maybe something could be done to expose this kind of error. :-)

#### Context

- axios version:  v. 0.15.0
- Environment: Node v6.11.1, Chrome Version 66.0.3359.139 (Official Build) (64-bit), windows 7

**Comment by Rain1368189893** [comment]
Yes， i also find this issues.
I find the way to slove the problem that: in axios/defaults.js and then change the code like this:
```
transformResponse: [function transformResponse(data) {
    /*eslint no-param-reassign:0*/
    if (typeof data === 'string') {
      try {
        if (!String.prototype.trim) {
          String.prototype.trim = function () {
            return this.replace(/^[\s\uFEFF\xA0]+|[\s\uFEFF\xA0]+$/g, '');
          };
        }
        data = JSON.parse(data.trim(''));
      } catch (e) { 
        /* Ignore */
        console.log('can not JSON.parse the response', e)
      }
    }
    return data;
  }],
```
hope this can help you!
ENV:
axios version: ^0.18.0
Node: v8.11.1
MAC

**Comment by chinesedfan** [comment]
Closed in favor of #61.
